src/Analyzers/CSharp/Analyzers/ConvertNameOf/CSharpConvertNameOfDiagnosticAnalyzer.cs

```diff
@@ -52,24 +53,24 @@ private void AnalyzeSyntaxAction(SyntaxNodeAnalysisContext syntaxContext)
 
             // TODO: Check for compiler errors on the declaration
 
-            var parent = node.Parent;
-
             // We know that it is a typeof() instance, but we only want to offer the fix if it is a .Name access
+            var parent = node.Parent;
             if (!(node.Parent.IsKind(SyntaxKind.SimpleMemberAccessExpression) && parent.IsNameMemberAccess()))
```

Comment:
Can we avoid walking the syntax parent and instead look at `IOperation.Parent` of this node? That should be an `IPropertyReferenceOperation` with it's property symbol named `FullName` for us to continue analysis. This will help you avoid calling `IsNameMemberAccess` as well and should be faster.